src/routing/error_cause.rs

```diff
@@ -198,12 +150,61 @@ impl From<&reqwest::Error> for ErrorCause {
     }
 }
 
-impl From<anyhow::Error> for ErrorCause {
-    fn from(e: anyhow::Error) -> Self {
-        if let Some(e) = error_infer::<AgentError>(&e) {
-            return Self::AgentError(e.to_string());
+// TODO update in `ic-boundary` to "canister_route_not_found" and then remove CANISTER_NOT_FOUND
+const CANISTER_NOT_FOUND: &str = "canister_not_found";
+const CANISTER_ROUTE_NOT_FOUND: &str = "canister_route_not_found";
+const SUBNET_NOT_FOUND: &str = "subnet_not_found";
+const NO_HEALTHY_NODES: &str = "no_healthy_nodes";
```

Comment:
as we have API nodes, I suggest a more explicit name here `NO_HEALTHY_REPLICA_NODES` or `NO_HEALTHY_REPLICA`

Reply:
The constants here are just so that we can use them in tests. Otherwise I'd just put the text directly. They just replicate the text in them.